src/frontend/js/hooks/usePendingContractCount/index.ts

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react';
+import useContractAbilities from 'hooks/useContractAbilities';
+import { useOrganizationContracts } from 'hooks/useContracts';
+import { useCourseProductRelation } from 'hooks/useCourseProductRelation';
+import { PER_PAGE } from 'settings';
+import { ContractState, CourseProductRelation, Organization } from 'types/Joanie';
+import { ContractActions } from 'utils/AbilitiesHelper/types';
+
+interface UsePendingContractCountProps {
+  organizationId?: Organization['id'];
+  courseProductRelationId?: CourseProductRelation['id'];
+}
+
+const useTeacherPendingContracts = ({
+  organizationId,
+  courseProductRelationId,
+}: UsePendingContractCountProps) => {
+  const {
+    item: training,
+    states: { isFetched: isTrainingFetched },
+  } = useCourseProductRelation(courseProductRelationId, {
+    organization_id: organizationId,
+  });
+
+  const isActive = useMemo(() => {
+    return !!(organizationId && (!courseProductRelationId || isTrainingFetched));
+  }, [organizationId, courseProductRelationId, isTrainingFetched]);
+
+  const { items: contracts, meta } = useOrganizationContracts(
+    {
+      organization_id: organizationId,
+      course_id: training?.course.id,
+      product_id: training?.product.id,
+      signature_state: ContractState.LEARNER_SIGNED,
+      page: 1,
+      page_size: PER_PAGE.teacherContractList,
+    },
+    { enabled: isActive },
+  );
+
+  const contractAbilities = useContractAbilities(contracts);
+
+  if (isActive) {
+    return {
+      contracts,
+      pendingContractCount: meta?.pagination?.count ?? 0,
+      canSignContracts: contractAbilities.can(ContractActions.SIGN),
+    };
+  }
+  return {
+    contracts: [],
+    pendingContractCount: 0,
+    canSignContracts: false,
+  };
```

Comment:
From my point of view, this is not the purpose of this hook to check if the authenticated user has the right to sign contract.